src/Utilities/CssDocument.php

```diff
@@ -82,23 +61,47 @@ public function __construct(string $css)
      */
     public function getStyleRulesData(array $allowedMediaTypes): array
     {
-        $splitCss = $this->splitCssAndMediaQuery($allowedMediaTypes);
-
         $ruleMatches = [];
-        foreach ($splitCss as $cssPart) {
-            // process each part for selectors and definitions
-            \preg_match_all('/(?:^|[\\s^{}]*)([^{]+){([^}]*)}/mi', $cssPart['css'], $matches, PREG_SET_ORDER);
-
-            foreach ($matches as $cssRule) {
+        /** @var CssRenderable $rule */
+        foreach ($this->sabberwormCssDocument->getContents() as $rule) {
+            if ($rule instanceof CssAtRuleBlockList && $rule->atRuleName() === 'media') {
+                /** @var string $mediaQueryList */
+                $mediaQueryList = $rule->atRuleArgs();
+                [$mediaType] = \explode('(', $mediaQueryList, 2);
+                if (\trim($mediaType) !== '') {
+                    $mediaTypesExpression = \implode('|', $allowedMediaTypes);
+                    if (!\preg_match('/^\\s*+(?:only\\s++)?+(?:' . $mediaTypesExpression . ')/i', $mediaType)) {
+                        continue;
+                    }
+                }
+                $media = '@media ' . $mediaQueryList;
+                /** @var CssRenderable $nestedRule */
+                foreach ($rule->getContents() as $nestedRule) {
```

Comment:
Would a decicated method in sabberworm/php-css-parser make sense for this (i.e., should we open a ticket)?

Reply:
> 
> 
> Would a decicated method in sabberworm/php-css-parser make sense for this (i.e., should we open a ticket)?

Looks like the wrong copypasta comment here.

I think it would be worth trying to get Psalm integrated into `sabberworm/php-css-parser`, and thus opening a ticket.

Although a year ago I was tearing my hair out with it (probably because of the caching issue which needs fixing in Psalm), it has very much grown on me since then.  It also paves the way forward for a defined standard for more complex DocBlock (and perhaps PHP itself) type annotations for which there have been rumbling discussions in various places for many years.

Reply:
> I think it would be worth trying to get Psalm integrated into `sabberworm/php-css-parser`, and thus opening a ticket.

At a minimum, it would be good to have proper type annotations on the public methods, both the basic PHP ones and the more detailed DocBlock ones that Psalm (and perhaps other tools) recognize.

It should also be easily possible to add Psalm to that project, and generate a current 'baseline', without making any actual code changes.

Reply:
I've actually started to add static type checking with PHPStan to `sabberworm/php-css-parser` and now am fixing the issues one by one: https://github.com/sabberworm/PHP-CSS-Parser/pull/227